tools/migration/migrate.go

```diff
@@ -0,0 +1,168 @@
+package migration
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/sync/errgroup"
+
+	"golang.org/x/mod/modfile"
+)
+
+// GoModUpdate defines a mapping of module path to the version it should be updated to.
+type GoModUpdate map[string]string
```

Comment:
```suggestion
type GoModUpdatePlan map[string]string
```

since this is like a plan for all of the mod replaces?